keras/backend/mxnet_backend.py

```diff
@@ -2915,7 +2910,54 @@ def conv1d(x, kernel, strides=1, padding='valid',
     # Returns
         A tensor, result of 1D convolution.
     """
-    raise NotImplementedError('MXNet Backend: conv1d is not supported yet.')
+    if data_format is None:
+        data_format = image_data_format()
+    _validate_data_format(data_format)
+
+    # Causal requires temporal padding.
+    # MXNet backend does not support temporal padding on 3D tensor.
+    if padding is 'causal':
+        raise ValueError('MXNet Backend: conv1d does not support "causal" padding mode')
+
+    if padding not in {'same', 'valid'}:
+        raise ValueError('`padding` should be either `same` or `valid`.')
+
+    if hasattr(x, '_keras_shape'):
+        shape = x._keras_shape
+    else:
+        shape = None
+
+    if data_format == 'channels_last':
+        # X original shape (batch, length, input_dim)
+        # Add a dimension to X to Make it (batch, length, 1, input_dim)
+        x = expand_dims(x, axis=2)
+        # update x._keras_shape
+        if shape is not None:
+            x._keras_shape = (shape[0], shape[1], 1, shape[2])
+    elif data_format == 'channels_first':
+        # X original shape (batch, input_dim, length)
+        # Add a dimension to X to make it (batch, input_dim, length, 1)
+        x = expand_dims(x, axis=3)
+        if shape is not None:
+            x._keras_shape = (shape[0], shape[1], shape[2], 1)
+
+    # update dilation rate, strides
+    dilation_rate = (dilation_rate, 1)
+    strides = (strides, 1)
+    # add dim to kernel (always same format independently of data_format)
+    # i.e. (rows, 1, input_depth, depth)
+    kernel = expand_dims(kernel, 1)
```

Comment:
Minor comment, `axis=1` instead of `1`, makes the code more readable.  Thanks!

Reply:
Thanks. Good call. Fixed for all.